test/Dialect/Torch/decompose-complex-ops.mlir

```diff
@@ -1332,3 +1332,17 @@ func.func @torch.aten.std.dim(%arg0: !torch.vtensor<[3,4,5],f32>) -> !torch.vten
   %0 = torch.aten.std.dim %arg0, %dims, %unbiased, %keepdim: !torch.vtensor<[3,4,5],f32>, !torch.list<int>, !torch.bool, !torch.bool -> !torch.vtensor<[3,4,1],f32>
   return %0 : !torch.vtensor<[3,4,1],f32>
 }
+
+// -----
+// CHECK-LABEL:  func.func @torch.aten.div.int(
+// CHECK-SAME:                            %[[LHS:.*]]: !torch.int,
+// CHECK-SAME:                            %[[RHS:.*]]: !torch.int) -> !torch.float {
+// CHECK:          %[[LHS_I64:.*]] = torch_c.to_i64 %[[LHS]]
+// CHECK:          %[[RHS_I64:.*]] = torch_c.to_i64 %[[RHS]]
+// CHECK:          %[[SUB:.*]] = arith.divf %[[LHS_I64:.*]], [[RHS_I64:.*]] : i64
+// CHECK:          %[[OUT:.*]] = torch_c.from_i64 %[[SUB:.*]]
+// CHECK:          return %[[OUT:.*]] : !torch.float
```

Comment:
This is not the right IR for the decomposition. The decomposition only creates `torch` dialect ops, but here you have `arith.divf` from the `arith` dialect. You can generate the right checks by using https://github.com/llvm/llvm-project/blob/2dde4ba63974daf59f8ce5c346505f194f920131/mlir/utils/generate-test-checks.py

Reply:
I get the following error when running the script you showed me ```bash: generate-test-checks.py: command not found```
Also, Prashant told me ```There is no lowering of atenFloatScalar op from torch to linalg.``` This could be a problem, but I see it's being used in ```ReifyShapeCalculations.cpp``` does this still mean lowering is needed?

Reply:
You need to call the script by doing `python generate-test-checks.py ...`. You can get more info on the script by passing the `-h` flag.

For the `AtenFloatScalar`, you should just need a `folder`, handling only the case where the input is a constant int, since your norm op only uses the `AtenDivInt` op to divide 1 by `p`, and `p` can be passed as a constant in the e2e tests

Reply:
You need to call the script as `python generate-test-checks.py ...` You can get more info on the script by passing the `-h` flag.

Looking a bit closer at what you need from `AtenDivInt` to get the norm op to work, I think all you need is the `fold` function and the tests for the folding. There is already a `fold` function for `AtenFloatScalar`. In your norm op all you need to do is make sure you hard-code the value for `p` rather than passing it as a parameter